src/cmds/qsub.c

```diff
@@ -472,15 +472,27 @@ expand_varlist(char *varlist)
 	char *p1, *p2, *p;
 	char *ev;
 	int  v_value1_sz=0;
+	char *pc;
+	int special_char_cnt = 0;
+
+	/*
+	 * count special characters as they are escaped with '\' in copy_env_value function
+	 * so that this is useful while calculating the accurate size of the destination string
+	 */
+	pc = varlist;
+	for(;*pc;pc++) {
+		if ((*pc == '"') || (*pc == '\'') || (*pc == ',') || (*pc == '\\'))
+			special_char_cnt++;
+	}
 
+	v_value1_sz = strlen(varlist) +  special_char_cnt + 1;
```

Comment:
If you are looping to count special characters anyway in the previous loop, why not do the string length right there and save a strlen here?

Reply:
Implemented.